### ISSUE TITLE
[BUG] V4 SDK hangs if Log level is set to DEBUG

### ISSUE THREAD

**Issue by moderakh** [Azure/azure-sdk-for-java#13143]
v4 4.1.0 SDK hangs with default configuration if Log level is set to DEBUG.

just run the following code against an existing database and container. `container.createItem(.)` will never succeed.

```java
CosmosAsyncClient client = new CosmosClientBuilder()
        .endpoint(TestConfigurations.HOST)
        .key(TestConfigurations.MASTER_KEY)
        .consistencyLevel(ConsistencyLevel.EVENTUAL)
        .buildAsyncClient();

// let's assume database and container already exist
CosmosAsyncContainer container = client.getDatabase(dbName).getContainer(containerName);

POJO pojo = new POJO();
pojo.id = UUID.randomUUID().toString();
pojo.prop = UUID.randomUUID().toString();

container.createItem(pojo).block();
```

**Comment by mayank-magoyal** [comment]
@xinlian12 is this backported to version 4.1?

**Comment by xinlian12** [comment]
@mayank-magoyal This fix is available >= 4.3, we did not backported to version 4.1
